packages/@aws-cdk/pipelines/lib/lambda/index.py

```diff
@@ -0,0 +1,35 @@
+import boto3
+import json
+
+cp = boto3.client('codepipeline')
+
+def handler(event, context):
+  pipeline_name = event['PipelineName']
+  stage_name = event['StageName']
+  action_name = stage_name + 'ManualApproval'
+
+  response = cp.get_pipeline_state(name=pipeline_name)
+
+  token = parse_pipeline_state(response, stage_name, action_name)
+
+  if token:
+    cp.put_approval_result(
+      pipelineName=pipeline_name,
+      stageName=stage_name,
+      actionName=action_name,
+      result={
+        'summary': 'No security changes detected. Automatically approved by Lambda.',
+        'status': 'Approved',
+      },
+      token=token
+    )
+
+def parse_pipeline_state(response, stageName, actionName):
+  stages = response['stageStates']
+  selected_stages = [s for s in stages if s['stageName'] == stageName]
+  if len(selected_stages) == 0:
+    return
+  manual_approval_action = [a for a in selected_stages[0]['actionStates'] if a['actionName'] == actionName]
+  if len(manual_approval_action) == 0:
+    return
+  return manual_approval_action[0]['latestExecution']['token']
```

Comment:
`latestExecution` makes me slightly uncomfortable. If we call this "too early", before the approval we actually want to approve, what are we going to see? Do we see the previous run's token? Or is the state of the action such that there is no `token` until the newest action will have started?

Consider the following scenario:

![image](https://user-images.githubusercontent.com/524162/124760816-683aaf00-df31-11eb-9d13-693051df75fd.png)

In **execution 13**, we haven't started the approval step yet, so does `latestExecution` return nothing? Or does it return the execution of that approval step in **execution 12** ? Probably the latter? Will it have a token? Probably not, but maybe it does.

I'd like you to investigate this, and/or reason about it, and reassure me with a source code comment that you've thought about the possible edge cases and that nothing will go wrong 🙂

Reply:
Okay yeah I'll dive deeper into this. But from previous testing and my other research is the following:
1. If this is the first time that the action has ever run: the action will not have a field "latestExecution" and this will return an empty object
2. If it has run before, it will return the last object:

I think a happy compromise that would fix the waiting issue from above is the following:

1. Record the timestamp upon entering the handler (`entryTimestamp`)  
2. Check to see if the status of the **latestExecution** is in progress. 
3. If it isn't wait 5 seconds to run the function `get_pipeline_state` again